### DISCUSSION TITLE
Does Zitadel provide support for some out of the box applicattions?

### DISCUSSION THREAD

**Discussion by vgtom** [zitadel/zitadel#1869] [Q&A] [answered]
Hi All,

Does zitadel provide support for some out of the box applications like gmail, outlook, google drive etc?
How do i integrate these apps in my project?

Thanks in advance
Vtom

**Comment by fforootd** [comment, marked as answer]
Hi @vgtom 

Thank you for the question. Right now ZITADEL does provide federation to existing identities with AzureAD, Google, and all OpenID Connect compliant IDPs. 
The other way around (to login with ZITADEL at e.g. M365) is not yet possible (due to the need for SAML). However this can be solved with a workaround.

What are you exactly looking for? We are interested to get to know you use-case!

Best

**Reply by gc-ss** [reply]
> Right now ZITADEL does provide federation to existing identities with AzureAD, Google, and all OpenID Connect compliant IDPs.

Very good to know. Are there some sample code we can look at and try out? Eg: Google/gmail and/or GitHub

**Reply by fforootd** [reply]
@gc-ss you don't need to code for this 😁 you can configure this on a ZITADEL system or organisation.

![image](https://user-images.githubusercontent.com/9879976/126983557-315615b8-b3a3-409f-a1f9-8769c9091756.png)

Some more insights to this can be found here -> https://docs.zitadel.ch/docs/guides/usage/identity-brokering

**Reply by mffap** [reply]
SAML is now available with the [release of v2.5.0](https://github.com/zitadel/zitadel/releases/tag/v2.5.0). 
You can find guides for [integrating Atlassian, Gitlab, AWS etc with ZITADEL in our docs](https://zitadel.com/docs/guides/integrate/services)
